frontend/deb/template_control.go

```diff
@@ -0,0 +1,153 @@
+package deb
+
+import (
+	_ "embed"
+	"fmt"
+	"io"
+	"strings"
+	"text/template"
+
+	"github.com/Azure/dalec"
+	"golang.org/x/exp/slices"
+)
+
+func WriteControl(spec *dalec.Spec, target string, w io.Writer) error {
+	return controlTmpl.Execute(w, &controlWrapper{spec, target})
+}
+
+type controlWrapper struct {
+	*dalec.Spec
+	Target string
+}
+
+func (w *controlWrapper) Architecture() string {
+	if w.NoArch {
+		return "all"
+	}
+	return "linux-any"
+}
+
+// appendConstraints takes an input list of packages and returns a new list of
+// packages with the constraints appended for use in a debian/control file.
+// The output list is sorted lexicographically.
+func appendConstraints(deps map[string]dalec.PackageConstraints) []string {
+	if deps == nil {
+		return nil
+	}
+	out := dalec.SortMapKeys(deps)
+
+	for i, dep := range out {
+		constraints := deps[dep]
+		s := dep
+		if len(constraints.Version) > 0 {
```

Comment:
Can we add a comment here that the format is specified here: https://www.debian.org/doc/debian-policy/ch-relationships.html and is `package (version constraint, ...) [arch constraint, ...]`

Reply:
Done